renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -1042,35 +1044,36 @@ bool WrappedVulkan::Serialise_vkEndCommandBuffer(SerialiserType &ser, VkCommandB
                  ToStr(GetResID(commandBuffer)).c_str());
 #endif
 
-        if(m_Partial[Primary].partialParent == BakedCommandBuffer &&
-           !m_RenderState.xfbcounters.empty())
+        VulkanRenderState &renderstate = GetCmdRenderState();
+        if(m_Partial[Primary].partialParent == BakedCommandBuffer && !renderstate.xfbcounters.empty())
         {
-          m_RenderState.EndTransformFeedback(commandBuffer);
+          renderstate.EndTransformFeedback(this, commandBuffer);
         }
 
         if(m_Partial[Primary].partialParent == BakedCommandBuffer &&
-           m_RenderState.IsConditionalRenderingEnabled())
+           renderstate.IsConditionalRenderingEnabled())
         {
-          m_RenderState.EndConditionalRendering(commandBuffer);
+          renderstate.EndConditionalRendering(commandBuffer);
         }
 
         // finish any render pass that was still active in the primary partial parent
         if(m_Partial[Primary].partialParent == BakedCommandBuffer &&
            m_Partial[Primary].renderPassActive)
         {
           uint32_t numSubpasses =
-              (uint32_t)m_CreationInfo.m_RenderPass[m_RenderState.renderPass].subpasses.size();
+              (uint32_t)m_CreationInfo.m_RenderPass[renderstate.renderPass].subpasses.size();
 
           // for each subpass we skip, and for the finalLayout transition at the end of the
           // renderpass, record these barriers. These are executed implicitly but because we want to
           // pretend they never happened, we then reverse their effects so that our layout tracking
           // is accurate and the images end up in the layout they were in during the last active
           // subpass
-          uint32_t &sub = m_BakedCmdBufferInfo[m_LastCmdBufferID].state.subpass;
+          // TODO: Why reference?
+          uint32_t sub = m_BakedCmdBufferInfo[m_LastCmdBufferID].state.subpass;
```

Comment:
I removed the reference, since otherwise it breaks. I don't really understand why the sub needed to be a reference, do you remember the reason? I have tested without the reference and everything seems to work, but I am not sure if I missed some use case.

Reply:
The reference previously was to ensure that the baked state's subpass was updated. I think it's fine to remove the reference now that the render state is tracked differently.